pkg/operator/controller/controller_dns.go

```diff
@@ -1,89 +1,139 @@
 package controller
 
 import (
+	"context"
 	"fmt"
 
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+
 	operatorv1 "github.com/openshift/api/operator/v1"
-	"github.com/openshift/cluster-ingress-operator/pkg/dns"
 
 	corev1 "k8s.io/api/core/v1"
 
-	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ensureDNS will create DNS records for the given LB service. If service is
 // nil, nothing is done.
-func (r *reconciler) ensureDNS(ci *operatorv1.IngressController, service *corev1.Service, dnsConfig *configv1.DNS) error {
-	records := desiredDNSRecords(ci, dnsConfig, service)
-	for _, record := range records {
-		err := r.DNSManager.Ensure(record)
-		if err != nil {
-			return fmt.Errorf("failed to ensure DNS record %v for %s/%s: %v", record, ci.Namespace, ci.Name, err)
-		}
-		log.Info("ensured DNS record for ingresscontroller", "namespace", ci.Namespace, "name", ci.Name, "record", record)
+func (r *reconciler) ensureWildcardDNSRecord(ic *operatorv1.IngressController, service *corev1.Service) (*iov1.DNSRecord, error) {
+	if service == nil {
+		return nil, nil
 	}
-	return nil
-}
 
-func newAliasRecord(domain, target string, zone configv1.DNSZone) *dns.Record {
-	return &dns.Record{
-		Zone: zone,
-		Type: dns.ALIASRecord,
-		Alias: &dns.AliasRecord{
-			Domain: domain,
-			Target: target,
-		},
+	desired := desiredWildcardRecord(ic, service)
+	current, err := r.currentWildcardDNSRecord(ic)
+	if err != nil {
+		return nil, err
 	}
-}
-
-func newARecord(domain, target string, zone configv1.DNSZone) *dns.Record {
-	return &dns.Record{
-		Zone: zone,
-		Type: dns.ARecordType,
-		ARecord: &dns.ARecord{
-			Domain:  domain,
-			Address: target,
-		},
+	if desired != nil && current == nil {
+		if err := r.client.Create(context.TODO(), desired); err != nil {
+			return nil, fmt.Errorf("failed to create dnsrecord %s/%s: %v", desired.Namespace, desired.Name, err)
+		}
+		log.Info("created dnsrecord", "dnsrecord", desired)
+		return desired, nil
 	}
```

Comment:
I am not sure we can do without update handling.  I would like to test the following on AWS:

1. Scale the default ingress controller to 1.
2. Look up the AZ of the sole ingress controller pod.
3. Kill ingress controller pods till one appears in a different AZ, which should cause the ELB's DNS name to change.
4. Check whether the wildcard DNS name still resolves.

Reply:
I performed the test, and the ELB's DNS name does not change; only the address to which it resolves changes, and because we use an alias on AWS, the wildcard DNS record likewise changes over to the new address without requiring an update by the operator.